cmd/maya-apiserver/cstor-operator/cspc/operations.go

```diff
@@ -187,3 +202,30 @@ func (pc *PoolConfig) getCSPWithNodeName(nodeName string) (*apis.NewTestCStorPoo
 	}
 	return nil, errors.New("No CSP(s) found")
 }
+
+// isBDUsable returns no error if BD can be used.
+// If BD has no BDC -- it is created
+// TODO: Move to algorithm package
+func (pc *PoolConfig) isBDUsable(bdName string) error {
+	bdObj, err := apisbd.
+		NewKubeClient().
+		WithNamespace(pc.AlgorithmConfig.Namespace).
+		Get(bdName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "could not get bd object %s", bdName)
+	}
+	err = pc.AlgorithmConfig.ClaimBD(bdObj)
+	if err != nil {
+		return errors.Wrapf(err, "failed to claim bd %s", bdName)
+
+	}
+	isBDUsable, err := pc.AlgorithmConfig.IsClaimedBDUsable(bdObj)
```

Comment:
In `IsClaimedBDUsable`, lets return err at the end of the function saying `BD claimed by other than <cspc-name>`

Reply:
Also, lets remove multi-return value from `IsClaimedBDUsable`.. lets just return `error`.. it avoids multiple checks in the caller